plugins/base/src/main/kotlin/renderers/html/NavigationDataProvider.kt

```diff
@@ -90,14 +91,17 @@ abstract class NavigationDataProvider {
         }
     }
 
+    private val navigationNodeOrder: Comparator<NavigationNode> =
+        compareBy(canonicalAlphabeticalOrder) { it.name }
+
     private fun ContentPage.navigableChildren() =
         if (this is ClasslikePage) {
             this.navigableChildren()
         } else {
             children
                 .filterIsInstance<ContentPage>()
                 .map { visit(it) }
-                .sortedBy { it.name.toLowerCase() }
+                .sortedWith(navigationNodeOrder)
```

Comment:
Could you please add a unit test for navigation to check the use case you described, in case the sorting ever changes

The test class is `NavigationTest`, it should be pretty straightforward